File: LeetCode/56. 合并区间.js
/**
 * @param {number[][]} intervals
 * @return {number[][]}
 */
var merge = function (intervals) {
  const n = intervals.length
  const res = []
  intervals.sort((a, b) => a[0] - b[0])
  let begin = intervals[0][0]
  let end = intervals[0][1]
  for (let i = 1; i < n; ++i) {
    if (intervals[i][0] > end) {
      res.push([begin, end])
      begin = intervals[i][0]
      end = intervals[i][1]
    } else {
      end = Math.max(end, intervals[i][1])
    }
  }
  res.push([begin, end])
  return res
}

File: LeetCode/1-100/56. 合并区间.java
class Solution {
    public int[][] merge(int[][] intervals) {
  Arrays.sort(intervals,(t1,t2) -> t1[0] - t2[0]);
        ArrayList<int[]> list = new ArrayList<>();
        int[] temp = intervals[0];
        for (int i = 1; i < intervals.length; i++) {
            if (intervals[i][0] > temp[1]) {
                list.add(temp);
                temp = intervals[i];
            } else {
                temp[1] = Math.max(intervals[i][1],temp[1]);
            }
        }
        list.add(temp);
        return list.toArray(new int[list.size()][]);
    }
}